rest_framework/fields.py

```diff
@@ -1381,7 +1381,10 @@ class ListField(Field):
     def __init__(self, *args, **kwargs):
         self.child = kwargs.pop('child', copy.deepcopy(self.child))
         self.allow_empty = kwargs.pop('allow_empty', True)
+
         assert not inspect.isclass(self.child), '`child` has not been instantiated.'
+        assert self.child.source is None, '`source=` should not be used on a `child`.'
```

Comment:
Perhaps?...

    "The source argument is not meaningful when applied to a `child=` field. Remove `source=` from the field declaration."